contrib/bind/provider.py

```diff
@@ -43,8 +43,8 @@ def remove_record(self, record):
 
     def first_setup(self, parser):
         # Insert SOA and NS records
-        hostname = unit_get('public-address')
-        addr = resolve_hostname_to_ip(hostname)
+        addr = unit_get('public-address')
+        #addr = resolve_hostname_to_ip(hostname)
```

Comment:
Tested on AWS

On Fri, Sep 25, 2015 at 6:16 PM, Charles Butler <notifications@github.com>
wrote:

> In contrib/bind/provider.py
> <https://github.com/chuckbutler/DNS-Charm/pull/26#discussion_r40447234>:
>
> > @@ -43,8 +43,8 @@ def remove_record(self, record):
> >
> >      def first_setup(self, parser):
> >          # Insert SOA and NS records
> > -        hostname = unit_get('public-address')
> > -        addr = resolve_hostname_to_ip(hostname)
> > +        addr = unit_get('public-address')
> > +        #addr = resolve_hostname_to_ip(hostname)
>
> Ok which substrate was this tested on? i'd like to run a series of
> validations across the clouds so we can set sail to that method then. Last
> i checked AWS was still giving the unit name, as well as single interface
> MAAS nodes.
>
> We can feesibly update that method to check with a regex, compare ipv4 vs
> alphanumeric to determine if we need to pass the data through the handler.
> Which seems like a slightly better resolution that maintains compatibility
> in the cases where it would fail.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/chuckbutler/DNS-Charm/pull/26/files#r40447234>.
>



-- 
Best Regards,
       Nicolas Thomas
http://insights.ubuntu.com/?p=889
EMEA Solution Architect              Canonical
GPG FPR: D592 4185 F099 9031 6590 6292 492F C740 F03A 7EB9